File: dss/common/Terminal.py
# Copyright (C) 2010 Richard Lincoln
#
# This library is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2.1 of the License, or (at your option) any later version.
#
# This library is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public
# License along with this library; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA, USA

class Terminal(object):
    """Each electrical element in the power system has one or more terminals.
    Each terminal has one or more conductors.  Each conductor contains a
    disconnect switch and a TCC (fuse) curve[Fuse has been disabled and is
    being redesigned; a Relay object can be used if needed to control the
    switches].  The conductors are numbered [1,2,3,...].  If the terminal is
    connected to an N-phase device, the first N conductors are assumed to
    correspond to the phases, in order.  The remaining conductors may be
    neutrals or whatever.
    """

    def __init__(self, busRef=0, termNodeRef=0, checked=False, nCond=0,
            activeConductor=0, conductors=None):
        """Initialises a new 'Terminal' instance.
        """
        self.busRef = busRef

        self.termNodeRef = termNodeRef

        self.checked = checked

        self.nCond = nCond

        self.activeConductor = activeConductor

        self.conductors = [] if conductors is None else conductors

File: core/dss/src/main/java/com/ncond/dss/common/Terminal.java
/**
 * Copyright (C) 2008-2012 Electric Power Research Institute, Inc.
 * Copyright (C) 2009-2012 Richard Lincoln
 * All rights reserved.
 */
package com.ncond.dss.common;

import lombok.Getter;
import lombok.Setter;

@Getter @Setter
public class Terminal {

	private int numCond;
	private int activeConductor;

	protected int busRef;

	protected int[] termNodeRef;
	protected Conductor[] conductors;
	protected boolean checked;

	public Terminal(int nCond) {
		super();
		numCond = nCond;
		busRef = -1;  // signify not set
		termNodeRef = new int[numCond];
		conductors = new Conductor[numCond];
		for (int i = 0; i < numCond; i++)
			conductors[i] = new Conductor();
		activeConductor = 0;
	}

	public void setActiveConductor(int value) {
		if (value >= 0 & value < numCond) activeConductor = value;
	}

	public Conductor getConductor(int idx) {
		return conductors[idx];
	}

	public int getTermNodeRef(int idx) {
		return termNodeRef[idx];
	}

	public int[] getTermNodeRef() {
		return termNodeRef;
	}

}
